src/main/java/io/Parser.java

```diff
@@ -0,0 +1,51 @@
+package io;
+import java.util.Dictionary;
+import java.util.Hashtable;
+import java.util.NoSuchElementException;
+/**
+ * Used parse the additional data
+ * given in a command based on tags.
+ * @author Alvis Ng (supermii2)
+ */
+public class Parser {
+    /** Untagged input*/
+    private String defaultInput;
+    /** Dictionary of tagged inputs*/
+    private Dictionary<String, String> taggedInputs;
+    /**
+     * Creates a parsed representation
+     * of the input string given by
+     * the user's input.
+     * @param input
+     */
+    public Parser(String input) {
+        taggedInputs = new Hashtable<>();
```

Comment:
Is there a reason why a hashtable is needed here? Since it seems like in all instances there will only be one element inside the hashtable. Would a simple array suffice?

Reply:
For Event, there are two inputs, from and to! This was my solution to preserving the input format given by the example, and essentially allows for inputs similar to kwargs in python! Not sure if it was necessary, but I think it can help futureproof in the event of a command with many different inputs.